marshmallow/fields.py

```diff
@@ -1348,8 +1363,19 @@ def _serialize(self, value, attr, obj, **kwargs):
                 if k in keys:
                     result[keys[k]] = v
         else:
-            for k, v in iteritems(value):
-                result[keys[k]] = self.value_container._serialize(v, None, None, **kwargs)
+            for key, val in iteritems(value):
+                try:
+                    deser_val = self.value_container._serialize(val, None, None, **kwargs)
+                except ValidationError as error:
+                    errors[key]['value'] = error.messages
+                    if error.valid_data is not None and key in keys:
+                        result[keys[key]] = error.valid_data
+                else:
+                    if key in keys:
+                        result[keys[key]] = deser_val
```

Comment:
I'm not really happy with copy and pasting this much code. It is effectively 6 copies of the same thing, with slight variations. I suspect there is a utility interface that can abstract this functionality while also improving the code's readability.

Reply:
I took a first pass at deduplicating the logic. I didn't see a clean way to abstract the operations across field classes, but I avoided introducing any duplicate logic in this PR and deduplicated a few operations that were already present.

I need to test what the performance implications are for this abstraction. It should me minimal, because the primary difference is `getattr` being used in place of static attribute access once in each method. An alternative would be to use a lambda instead, but I don't have an intuition for which is preferable or faster.

A few methods were passing `None` into  the `attr` and `obj`/`data` arguments instead of passing them through like the rest. Normalizing this did not affect the tests, but I suspect the original behavior was an obscure bug. I need to look at the implications of this change a little closer.

Reply:
> I suspect the original behavior was an obscure bug.

Confirmed. See #1176.